src/test/java/org/opentripplanner/middleware/utils/ItineraryUtilsTest.java

```diff
@@ -77,70 +117,99 @@ public void tearDownAfterTest() {
+        }
+
         // FIXME: For now, just check that the first itinerary in the list is valid. If we expand our check window from
         //  7 days to 14 (or more) days, this may need to be adjusted.
-        Assertions.assertTrue(existence.monday.isValid());
-        Assertions.assertTrue(ItineraryUtils.itinerariesMatch(expectedItinerary, existence.monday.itineraries.get(0)));
-        Assertions.assertTrue(existence.tuesday.isValid());
-        Assertions.assertTrue(ItineraryUtils.itinerariesMatch(expectedItinerary, existence.tuesday.itineraries.get(0)));
-        Assertions.assertTrue(existence.thursday.isValid());
-        Assertions.assertTrue(
-            ItineraryUtils.itinerariesMatch(expectedItinerary, existence.thursday.itineraries.get(0))
-        );
-        Assertions.assertTrue(existence.saturday.isValid());
-        Assertions.assertTrue(
-            ItineraryUtils.itinerariesMatch(expectedItinerary, existence.saturday.itineraries.get(0))
-        );
-        Assertions.assertTrue(existence.sunday.isValid());
-        Assertions.assertTrue(ItineraryUtils.itinerariesMatch(expectedItinerary, existence.sunday.itineraries.get(0)));
+        for (ItineraryExistence.ItineraryExistenceResult validDay : validDays) {
+            Assertions.assertTrue(validDay.isValid());
+            Assertions.assertTrue(ItineraryUtils.itinerariesMatch(expectedItinerary, validDay.itineraries.get(0)));
+        }
 
+        // When the check is not requested for a day, the existence entry will be null.
         Assertions.assertNull(existence.wednesday);
         Assertions.assertNull(existence.friday);
     }
 
+    private static Stream<Arguments> createCheckAllItinerariesExistTestCases() {
+        return Stream.of(
+            Arguments.of(false, "checkAllDays = false should produce allCheckedDaysAreValid = true."),
+            Arguments.of(true, "checkAllDays = true should produce allCheckedDaysAreValid = false.")
+        );
+    }
+
     /**
-     * Test case in which at least one itinerary does not exist,
-     * and therefore result.allCheckedDatesAreValid should be false.
+     * @return The new {@link Date} object with the date portion set to the specified {@link LocalDate} in OTP timezone.
      */
-    @Test
-    public void canCheckAtLeastOneTripDoesNotExist() throws URISyntaxException {
-        MonitoredTrip trip = makeTestTrip();
+    public static Date getNewItineraryDate(Date itineraryDate, LocalDate date) {
+        Instant startInstant = itineraryDate.toInstant();
 
-        // Set mocks to a list of responses, one without an itinerary.
-        OtpResponse resp = otpDispatcherPlanResponse.getResponse();
-        TestUtils.setupOtpMocks(List.of(resp, resp, resp, otpDispatcherPlanErrorResponse.getResponse(), resp));
+        return Date.from(LocalDateTime.of(
+            date,
+            LocalTime.ofInstant(startInstant, OTP_ZONE_ID)
+        ).toInstant(OTP_ZONE_ID.getRules().getOffset(startInstant)));
```

Comment:
I'm not a huge fan of using offsets. I suppose this might be safe to use since the offset is being retrieved for a certain point in time. But here is an alternative implementation of this:

```java
return new Date(
    ZonedDateTime.ofInstant(itineraryDate.toInstant(), DateTimeUtils.getOtpZoneId())
        .withYear(date.getYear())
        .withMonth(date.getMonthValue())
        .withDayOfMonth(date.getDayOfMonth())
        .toInstant()
        .toEpochMilli()
);
```

Reply:
Updated in https://github.com/ibi-group/otp-middleware/pull/102/commits/98f91c7d7043e0039a39a1855dc3e40befbf178e. I agree, the offset and rule thing feels complicated, and I was trying to key in the date parts, but there is a sleek syntax using `.with(date)` that takes care of it.